lib/Elastica/AbstractScript.php

```diff
@@ -1,21 +1,36 @@
 <?php
 namespace Elastica;
 
+use Elastica\Exception\InvalidException;
+
 /**
  * Base class for Script object.
  *
  * @author Nicolas Assing <nicolas.assing@gmail.com>
+ * @author avasilenko <aa.vasilenko@gmail.com>
  *
  * @link http://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting.html
  */
 abstract class AbstractScript extends AbstractUpdateAction
 {
+    const LANG_MVEL = 'mvel';
+    const LANG_JS = 'js';
+    const LANG_GROOVY = 'groovy';
+    const LANG_PYTHON = 'python';
+    const LANG_NATIVE = 'native';
+
+    protected $script;
+
+    protected $lang;
+
     /**
      * @param array|null $params
      * @param string     $id
      */
-    public function __construct(array $params = null, $id = null)
+    public function __construct($script, array $params = null, $lang = null, $id = null)
```

Comment:
If I understand this right, this is only a BC break for people that extended from AbstractScript. For all the inherited classes there is no difference that is not part of the library?